---
title: "TNF - AD Mendelian Randomization"
author: "Shea Andrews"
date: "6/6/2019"
output:
  html_document:
    df_print: paged
    keep_md: false
    toc: true
    toc_depth: 3
    number_sections: false
    toc_float:
      collapsed: false
      smooth_scroll: true
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE, echo=F}
## setwd('')
library(TwoSampleMR)
library(tidyverse)
source('scripts/PowerFunctions.R')
```

```{r, read data, include=TRUE, cache=T, echo=F, message=F, warning = FALSE}

tnfqtl <- 'input/tnf_qtls.txt' %>% 
  read_table2()

## load
load <- 'data/Kunkle2019load.chrall.CPRA_b37.tsv.gz' %>% 
  read_table2(., comment = '##', col_types = 'cddccdcddddddddccccccc')  %>%
  select(ID, CHROM, POS, REF, ALT, AF, TRAIT, BETA, SE, Z, P, N, DBSNP_ID, OLD_ID)

## RA
artis <- 'data/Okada2014rartis.chrall.CPRA_b37.tsv.gz' %>% 
  read_table2(., comment = '##', col_types = 'cddccdcddddddddccccccc')  %>%
  select(ID, CHROM, POS, REF, ALT, AF, TRAIT, BETA, SE, Z, P, N, DBSNP_ID, OLD_ID)

tnfa <- 'data/Sliz2019tnfa.chrall.CPRA_b37.tsv.gz' %>% 
  read_table2(., comment = '##', col_types = 'cddccdcddddddddddcdcccccc')  %>%
  select(ID, CHROM, POS, REF, ALT, AF, TRAIT, BETA, SE, Z, P, N, DBSNP_ID, OLD_ID) %>% 
  filter(P < 1e-5)

## RA - LOAD
ra_load.mrdat <- read_csv('input/Okada2014rartis_5e-8_Kunkle2019load_MRdat.csv') %>% 
  select(-pt)  %>% 
  mutate(pve.exposure = snp.pve(eaf.exposure, beta.exposure, se.exposure, samplesize.exposure))

# ## TNFa - LOAD
# tnfa_load.mrdat <- read_csv('input/AholaOlli2018tnfa_5e-6_Kunkle2019load_MRdat.csv') %>% 
#   select(-pt)  %>% 
#   mutate(pve.exposure = snp.pve(eaf.exposure, beta.exposure, se.exposure, samplesize.exposure))

```

# Introduction 
Epidemiological research has suggested that inhibition of tumor necrosis factor (TNF)-α in patients with rheumatoid arthritis (RA) reduces the overall risk of Alzheimer’s disease (AD). TNF-α antagonists have been suggested as a potential treatment for AD. In this study, we use Mendelian randomization to evaluate if RA, TNF gene expression and TNF-α levels are causally related to AD risk.

# Exposures
## TNF expression QTLs

Unraveling the polygenic architecture of complex traits using blood eQTL metaanalysis. [Võsa U et al. bioRxiv. 2018:447367](https://www.biorxiv.org/content/10.1101/447367v1)

Significant cis-eQTLs (FDR < 0.05) for TNF in whole blood after clumping 
```{r, echo=FALSE, message=F, warning = FALSE}
## TNF cis-eqtl
tnfceqtl_exposure.dat <- tnfqtl %>% 
  filter(TRAIT == 'TNF' & QTL == 'cis_eQTL') %>% 
  #filter(P < 5e-8) %>% 
  mutate(AF = ifelse(DBSNP_ID == 'rs9469017', 1-AF, AF)) %>%
  format_data(., type = 'exposure',
                               snp_col = 'DBSNP_ID',
                               beta_col = "BETA",
                               se_col = "SE",
                               eaf_col = "AF",
                               effect_allele_col = "ALT",
                               other_allele_col = "REF",
                               pval_col = "P",
                               z_col = "Z",
                               samplesize_col = "N",
                               phenotype_col = 'TRAIT')

tnfceqtl_exposure.clmpdat <- clump_data(tnfceqtl_exposure.dat)

tnfqtl %>%
  filter(TRAIT == 'TNF' & QTL == 'cis_eQTL') %>% 
  select(DBSNP_ID, CHROM, POS) %>%
  right_join(tnfceqtl_exposure.clmpdat, by = c('DBSNP_ID' = 'SNP')) %>% 
  rename(SNP = DBSNP_ID) %>%
  select(SNP, CHROM, POS, other_allele.exposure, effect_allele.exposure, eaf.exposure, beta.exposure, se.exposure, z.exposure, pval.exposure, samplesize.exposure, exposure, mr_keep.exposure) %>%
  as_data_frame()
```

## TNFa proteomic QTLs

Sliz, Eeva,  et al. 2019. “Genome-Wide Association Study Identifies Seven Novel Loci Associating with Circulating Cytokines and Cell Adhesion Molecules in Finns.” Journal of Medical Genetics, June. https://doi.org/10.1136/jmedgenet-2018-105965.
Genome-wide association study identifies seven novel loci associating with 2 circulating cytokines and cell adhesion molecules in Finns. [Silz et al. BioRxiv 2018](https://www.biorxiv.org/content/10.1101/491852v1). 

* Conducted a meta-analyssi with the data from Ahola-Olli et al 2017. One SNP was significantly associated with TNFa.

Significant TNFa pQTLs after clumping 
```{r, echo=FALSE, message=F, warning = FALSE}
# Formating and clumping exposure
## TNF pQTL
tnfa_exposure.dat <- tnfa %>% 
  filter(P < 5e-6) %>% 
  format_data(., type = 'exposure',
                               snp_col = 'DBSNP_ID',
                               beta_col = "BETA",
                               se_col = "SE",
                               eaf_col = "AF",
                               effect_allele_col = "ALT",
                               other_allele_col = "REF",
                               pval_col = "P",
                               z_col = "Z",
                               samplesize_col = "N",
                               phenotype_col = 'TRAIT')

tnfa_exposure.clmpdat <- clump_data(tnfa_exposure.dat) %>% 
  ## Add rs17074575 as it is removed from clumpping analysis 
  bind_rows(filter(tnfa_exposure.dat, SNP == 'rs17074575'))

tnfa %>%
  select(DBSNP_ID, CHROM, POS) %>%
  right_join(tnfa_exposure.clmpdat, by = c('DBSNP_ID' = 'SNP')) %>% 
  rename(SNP = DBSNP_ID) %>%
  select(SNP, CHROM, POS, other_allele.exposure, effect_allele.exposure, eaf.exposure, beta.exposure, se.exposure, z.exposure, pval.exposure, samplesize.exposure, exposure, mr_keep.exposure) %>%
  as_data_frame()
```

# MR analysis 
```{r, echo=F}
samplesize <- tibble(trait = c('AD', 'AAOS', 'RA'), 
                     samplesize = c(63926, 40255, 58284), 
                     ncase = c(21982, 14406, 14361), 
                     ncontrol = c(41944, 25849, 43923), 
                     prop = ncase / samplesize)
```


## TNF eQTL

### LOAD 

Extract exposure SNPs from LOAD GWAS and harmonize exposure and outcome datasets
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_load.dat <- load %>% 
  semi_join(tnfceqtl_exposure.clmpdat, by = c('DBSNP_ID' = 'SNP')) %>%
  format_data(., type = 'outcome',
                              snp_col = 'DBSNP_ID',
                              beta_col = "BETA",
                              se_col = "SE",
                              eaf_col = "AF",
                              effect_allele_col = "ALT",
                              other_allele_col = "REF",
                              pval_col = "P",
                              z_col = "Z",
                              samplesize_col = "N",
                              phenotype_col = 'TRAIT')

tnfceqtl_load.mrdat <- harmonise_data(tnfceqtl_load.dat, tnfceqtl_exposure.clmpdat) %>% 
  mutate(pve.exposure = snp.pve(eaf.exposure, beta.exposure, se.exposure, samplesize.exposure))  

tnfceqtl_load.mrdat %>% 
  as_data_frame()
```

Power analysis 
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_load.pve <- tnfceqtl_load.mrdat %>% 
  mutate(pve.exposure = snp.pve(eaf.exposure, beta.exposure, se.exposure, samplesize.exposure)) %>% 
  summarise(pve.exposure = sum(pve.exposure), samplesize.exposure = max(samplesize.exposure), samplesize.outcome = max(samplesize.outcome), nsnps = n()) 

tnfceqtl_load.power <- with(tnfceqtl_load.pve, results_binary(samplesize.exposure, samplesize.outcome, nsnps, 0.05, pve.exposure, 0.34, 1.1)) 

tnfceqtl_load.power %>% as_data_frame()
```

MR analysis 
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_load.mrres <- mr(tnfceqtl_load.mrdat, method_list = c("mr_ivw_fe", "mr_weighted_median", "mr_egger_regression")) 

tnfceqtl_load.res <- mr_singlesnp(tnfceqtl_load.mrdat, all_method = c("mr_ivw_fe", "mr_weighted_median", "mr_egger_regression")) 

tnfceqtl_load.res %>% 
  select(exposure, outcome, SNP, b, se, p) %>% 
  as_data_frame() 

```

Heterogenity Tests
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_load.het <- mr_heterogeneity(tnfceqtl_load.mrdat, method_list=c("mr_egger_regression", "mr_ivw")) 

tnfceqtl_load.het %>% 
  select(exposure, outcome, method, Q, Q_df, Q_pval, Q_pval) %>% 
  as_data_frame() 
```

plots
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_load.p <- mr_scatter_plot(tnfceqtl_load.mrres, tnfceqtl_load.mrdat) %>% 
  .[[1]] + theme_bw() + theme(legend.position = 'bottom')
tnfceqtl_load.p
```

### Rheumatoid Arthritis 

Extract exposure SNPs from RA GWAS and harmonize exposure and outcome datasets
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_artis.dat <- artis %>% 
  semi_join(tnfceqtl_exposure.clmpdat, by = c('DBSNP_ID' = 'SNP')) %>%
  format_data(., type = 'outcome',
                              snp_col = 'DBSNP_ID',
                              beta_col = "BETA",
                              se_col = "SE",
                              eaf_col = "AF",
                              effect_allele_col = "ALT",
                              other_allele_col = "REF",
                              pval_col = "P",
                              z_col = "Z",
                              samplesize_col = "N",
                              phenotype_col = 'TRAIT')

tnfceqtl_artis.mrdat <- harmonise_data(tnfceqtl_artis.dat, tnfceqtl_exposure.clmpdat) %>% 
  mutate(pve.exposure = snp.pve(eaf.exposure, beta.exposure, se.exposure, samplesize.exposure))  

tnfceqtl_artis.mrdat %>% 
  as_data_frame()
```

Power analysis 
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_artis.power <- tnfceqtl_artis.mrdat %>% 
  mutate(pve.exposure = snp.pve(eaf.exposure, beta.exposure, se.exposure, samplesize.exposure)) %>% 
  summarise(pve.exposure = sum(pve.exposure), samplesize.exposure = max(samplesize.exposure), samplesize.outcome = max(samplesize.outcome), nsnps = n()) 

with(tnfceqtl_artis.power, results_binary(samplesize.exposure, samplesize.outcome, nsnps, 0.05, pve.exposure, 0.24, 1.12)) %>% as_data_frame()
```

MR analysis 
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_artis.res <- mr_singlesnp(tnfceqtl_artis.mrdat, all_method = c("mr_ivw_fe", "mr_weighted_median", "mr_weighted_mode","mr_egger_regression")) 

tnfceqtl_artis.mrres <- mr(tnfceqtl_artis.mrdat, method_list = c("mr_ivw_fe", "mr_weighted_median", "mr_egger_regression")) 

tnfceqtl_artis.res %>% 
  select(exposure, outcome, SNP, b, se, p) %>% 
  as_data_frame() 
```

Heterogenity Tests
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_artis.het <- mr_heterogeneity(tnfceqtl_artis.mrdat, method_list=c("mr_egger_regression", "mr_ivw")) 

tnfceqtl_artis.het %>% 
  select(exposure, outcome, method, Q, Q_df, Q_pval, Q_pval) %>% 
  as_data_frame() 
```

plots
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_artis.p <- mr_scatter_plot(tnfceqtl_artis.mrres, tnfceqtl_artis.mrdat) %>% 
  .[[1]] + theme_bw() + theme(legend.position = 'bottom')
tnfceqtl_artis.p
```

## Rheumatoid Arthritis 

### LOAD
Power analysis 
```{r, echo=FALSE, message=F, warning = FALSE}
ra_load.power <- ra_load.mrdat %>% 
  mutate(pve.exposure = snp.pve(eaf.exposure, beta.exposure, se.exposure, samplesize.exposure)) %>% 
  summarise(pve.exposure = sum(pve.exposure), samplesize.exposure = max(samplesize.exposure), samplesize.outcome = max(samplesize.outcome), nsnps = n()) 

with(ra_load.power, results_binary(samplesize.exposure, samplesize.outcome, nsnps, 0.05, pve.exposure, 0.34, 1.055)) %>% as_data_frame()
```

MR analysis 
```{r, echo=FALSE, message=F, warning = FALSE}
ra_load.res <- mr(ra_load.mrdat, method_list = c("mr_ivw_fe", "mr_weighted_median", "mr_egger_regression")) 
ra_load.single <- mr_singlesnp(ra_load.mrdat, all_method = c("mr_ivw_fe", "mr_weighted_median", "mr_egger_regression")) 

ra_load.res %>% 
  select(exposure, outcome, method, nsnp, b, se, pval) %>% 
  as_data_frame() 
```

Heterogenity Tests
```{r, echo=FALSE, message=F, warning = FALSE}
ra_load.het <- mr_heterogeneity(ra_load.mrdat, method_list=c("mr_egger_regression", "mr_ivw")) 

ra_load.het %>% 
  select(exposure, outcome, method, Q, Q_df, Q_pval, Q_pval) %>% 
  as_data_frame() 
```

## TNF-a 

### LOAD
Extract TNFa exposure SNPs from LOAD GWAS and harmonize exposure and outcome datasets
```{r, echo=FALSE, message=F, warning = FALSE}
tnfapqtl_load.dat <- load %>% 
  semi_join(tnfa_exposure.clmpdat, by = c('DBSNP_ID' = 'SNP')) %>%
  format_data(., type = 'outcome',
                              snp_col = 'DBSNP_ID',
                              beta_col = "BETA",
                              se_col = "SE",
                              eaf_col = "AF",
                              effect_allele_col = "ALT",
                              other_allele_col = "REF",
                              pval_col = "P",
                              z_col = "Z",
                              samplesize_col = "N",
                              phenotype_col = 'TRAIT')

tnfa_load.mrdat <- harmonise_data(tnfapqtl_load.dat, tnfa_exposure.clmpdat) %>% 
  mutate(pve.exposure = snp.pve(eaf.exposure, beta.exposure, se.exposure, samplesize.exposure))  

tnfa_load.mrdat %>% 
  as_data_frame()
```

Power analysis 
```{r, echo=FALSE, message=F, warning = FALSE}
tnfa_load.power <- tnfa_load.mrdat %>% 
  summarise(pve.exposure = sum(pve.exposure), samplesize.exposure = max(samplesize.exposure), samplesize.outcome = max(samplesize.outcome), nsnps = n())

with(tnfa_load.power, results_binary(samplesize.exposure, samplesize.outcome, nsnps, 0.05, pve.exposure, 0.34, 1.190)) %>% as_data_frame()
```

MR analysis 
```{r, echo=FALSE, message=F, warning = FALSE}
tnfa_load.res <- mr(tnfa_load.mrdat, method_list = c("mr_ivw_fe", "mr_weighted_median", "mr_egger_regression")) 
tnfa_load.single <- mr_singlesnp(tnfa_load.mrdat, all_method = c("mr_ivw_fe", "mr_weighted_median", "mr_egger_regression")) 

tnfa_load.res %>% 
  select(exposure, outcome, method, nsnp, b, se, pval) %>% 
  as_data_frame() 
```

Heterogenity Tests
```{r, echo=FALSE, message=F, warning = FALSE}
tnfa_load.het <- mr_heterogeneity(tnfa_load.mrdat, method_list=c("mr_egger_regression", "mr_ivw")) 

tnfa_load.het %>% 
  select(exposure, outcome, method, Q, Q_df, Q_pval, Q_pval) %>% 
  as_data_frame() 
```

## Results

Table 1
```{r, echo=FALSE, message=F, warning = FALSE}

tnfceqtl_load.out <- generate_odds_ratios(tnfceqtl_load.res) %>% 
  mutate_at(c('b', 'se', 'p',  'lo_ci', 'up_ci', 'or', 'or_lci95', 'or_uci95'), round, 2) %>% 
  mutate(b_out = paste0(b, ' (', se, ')'), or_out = paste0(or, ' (', or_lci95, ', ', or_uci95, ')')) %>% 
  select(exposure, SNP, b_out, or_out, p) %>% 
  as_tibble() %>% 
  mutate(SNP = str_replace(SNP, 'All - Inverse variance weighted \\(fixed effects\\)', 'IVW')) %>% 
  mutate(SNP = str_replace(SNP, 'All - Weighted median', 'WME')) %>% 
  mutate(SNP = str_replace(SNP, 'All - MR Egger', 'MR Egger')) %>% 
  rename(method = SNP, pval = p)

tnfa_load.out <- generate_odds_ratios(tnfa_load.res) %>% 
  mutate_at(c('b', 'se', 'pval',  'lo_ci', 'up_ci', 'or', 'or_lci95', 'or_uci95'), round, 2) %>% 
  mutate(b_out = paste0(b, ' (', se, ')'), or_out = paste0(or, ' (', or_lci95, ', ', or_uci95, ')')) %>% 
  select(exposure, method, b_out, or_out, pval)  %>% 
  mutate(method = str_replace(method, 'Inverse variance weighted \\(fixed effects\\)', 'IVW')) %>% 
  mutate(method = str_replace(method, 'Weighted median', 'WME')) %>% 
  as_tibble()

ra_load.out <- generate_odds_ratios(ra_load.res) %>% 
  mutate_at(c('b', 'se', 'pval',  'lo_ci', 'up_ci', 'or', 'or_lci95', 'or_uci95'), round, 2) %>% 
  mutate(b_out = paste0(b, ' (', se, ')'), or_out = paste0(or, ' (', or_lci95, ', ', or_uci95, ')')) %>% 
  select(exposure, method, b_out, or_out, pval)  %>% 
  mutate(method = str_replace(method, 'Inverse variance weighted \\(fixed effects\\)', 'IVW')) %>% 
  mutate(method = str_replace(method, 'Weighted median', 'WME')) %>% 
  as_tibble()


tnfceqtl_load.out %>% 
  bind_rows(tnfa_load.out) %>% 
  bind_rows(ra_load.out) %>% 
  mutate(exposure = str_replace(exposure, 'Okada2014rartis', 'RA')) %>% 
  mutate(exposure = str_replace(exposure, 'AholaOlli2018tnfa', 'TNFa')) %>%
  write_csv('output/table1.csv')
```


Supplementary Table 1
```{r, echo=FALSE, message=F, warning = FALSE}
tnfceqtl_load.mrdat %>% 
  bind_rows(tnfa_load.mrdat) %>% 
  bind_rows(ra_load.mrdat) %>% 
  mutate(exposure = str_replace(exposure, 'Okada2014rartis', 'RA')) %>% 
  mutate(exposure = str_replace(exposure, 'AholaOlli2018tnfa', 'TNFa')) %>%
  mutate(outcome = 'LOAD') %>%
  write_csv('output/ST1.csv')


```













